internal/urlutil.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright (C)  2018 Yahoo Japan Corporation Athenz team.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package urlutil contains the utility functions for URL processing
+package urlutil
+
+import (
+	"errors"
+	"net/url"
+)
+
+var (
+	// ErrUnsupportedScheme is "Unsupported scheme, only support HTTP or HTTPS"
+	ErrUnsupportedScheme = errors.New("Unsupported scheme, only support HTTP or HTTPS")
+)
+
+// TrimHTTPScheme check and trim the URL scheme
+func TrimHTTPScheme(urlStr string) (string, error) {
+	url, err := url.Parse(urlStr)
+	if err != nil {
+		return "", err
+	}
+
+	// check scheme
+	scheme := url.Scheme
+	if scheme != "" {
+		if scheme != "http" && scheme != "https" {
+			return "", ErrUnsupportedScheme
+		}
+	}
+
+	// trim scheme, and remove '//' prefix
+	url.Scheme = ""
+	if url.Host == "" {
```

Comment:
This line causes a bug.
url.Scheme gets the string before the first occurrence of `:`.
https://golang.org/src/net/url/url.go?s=12320:12452#L445

Therefore, when like `http:ftp://example.com/path` inputed, `ftp://example.com/path` will return.

I think that is better to if url.Host is null then return error.

Reply:
It is for backward compatibility... The `"example.com/path"` case
Seems we cannot use the `net/url`, change back to regex?

```go
package main

import (
	"fmt"
	"net/url"
)

func main() {
	u1, _ := url.Parse("http:ftp://example.com/path")
	u2, _ := url.Parse("http://example.com/path")
	u3, _ := url.Parse("http://http://example.com/path")
	u4, _ := url.Parse("example.com/path")
	
	fmt.Printf("%+v\n", *u1)
	fmt.Printf("%+v\n", *u2)
	fmt.Printf("%+v\n", *u3)
	fmt.Printf("%+v\n", *u4)
}
```

output:
```
{Scheme:http Opaque:ftp://example.com/path User: Host: Path: RawPath: ForceQuery:false RawQuery: Fragment:}
{Scheme:http Opaque: User: Host:example.com Path:/path RawPath: ForceQuery:false RawQuery: Fragment:}
{Scheme:http Opaque: User: Host:http: Path://example.com/path RawPath: ForceQuery:false RawQuery: Fragment:}
{Scheme: Opaque: User: Host: Path:example.com/path RawPath: ForceQuery:false RawQuery: Fragment:}
```

Reply:
@No1zy  I think it is expected. We cannot and shouldn't handle all the cases for users.

Reply:
@kevindiu @WindzCUHK  
OK, I got it. However, still there is this problem.
https://github.com/yahoojapan/athenz-authorizer/pull/25#discussion_r309522099

url.Parse can not get host when a string such as `example.com: 8080` is input.
I think it is better to revert to the previous implementation.

Reply:
reverted